Kickstarter-iOS/Views/Controllers/BetaToolsViewController.swift

```diff
@@ -5,119 +5,169 @@ import Prelude
-      AppEnvironment.current.apiService.serverConfig.environment.rawValue
+    self.viewModel.outputs.goToPushNotificationTools
+      .observeForControllerAction()
+      .observeValues { [weak self] in
+        self?.goToDebugPushNotifications()
+      }
 
-    _ = self.environmentTitleLabel
-      |> settingsTitleLabelStyle
-  }
+    self.viewModel.outputs.goToFeatureFlagTools
+      .observeForControllerAction()
+      .observeValues { [weak self] in
+        self?.goToFeatureFlagTools()
+      }
 
-  override func bindViewModel() {
-    self.environmentSwitcher.rac.title = self.viewModel.outputs.environmentSwitcherButtonTitle
+    self.viewModel.outputs.goToBetaFeedback
+      .observeForControllerAction()
+      .observeValues { [weak self] in
+        self?.goToBetaFeedback()
+      }
 
-    self.languageSwitcher.rac.title = self.viewModel.outputs.currentLanguage
-      .map { $0.displayString }
+    self.viewModel.outputs.showChangeEnvironmentSheetWithSourceViewIndex
+      .observeForControllerAction()
+      .observeValues { [weak self] index in
+        self?.showEnvironmentActionSheet(sourceViewIndex: index)
+      }
 
-    self.viewModel.outputs.currentLanguage
-      .observeForUI()
-      .observeValues { [weak self] language in self?.languageDidChange(language: language) }
+    self.viewModel.outputs.showChangeLanguageSheetWithSourceViewIndex
+      .observeForControllerAction()
+      .observeValues { [weak self] index in
+        self?.showLanguageActionSheet(sourceViewIndex: index)
+      }
 
-    self.viewModel.outputs.goToBetaFeedback
+    self.viewModel.outputs.updateLanguage
       .observeForControllerAction()
-      .observeValues { [weak self] in self?.goToBetaFeedback() }
+      .observeValues { [weak self] language in
+        self?.updateLanguage(language: language)
+      }
 
-    self.viewModel.outputs.betaFeedbackMailDisabled
+    self.viewModel.outputs.updateEnvironment
       .observeForControllerAction()
-      .observeValues { [weak self] in self?.showMailDisabledAlert() }
+      .observeValues { [weak self] environment in
+        self?.updateEnvironment(environment: environment)
+      }
 
     self.viewModel.outputs.logoutWithParams
       .observeForControllerAction()
-      .observeValues { [weak self] in self?.logoutAndDismiss(params: $0) }
+      .observeValues { [weak self] in
+        self?.logoutAndDismiss(params: $0)
+      }
+
+    self.viewModel.outputs.showMailDisabledAlert
+      .observeForControllerAction()
+      .observeValues { [weak self] in
+        self?.showMailDisabledAlert()
+      }
   }
 
-  @IBAction fileprivate func betaFeedbackButtonTapped(_: Any) {
+  // MARK: - Selectors
+
+  @objc func betaFeedbackButtonTapped() {
     self.viewModel.inputs.betaFeedbackButtonTapped(canSendMail: MFMailComposeViewController.canSendMail())
   }
 
-  @IBAction fileprivate func betaDebugPushNotificationsButtonTapped(_: Any) {
+  @objc func doneTapped() {
```

Comment:
`private`

and also are we not routing this through VM for simplicity?

Reply:
Also if we pass this action from outside this controller would not need to know about how to dismiss itself (so no matter if we present it modally or on a navigation stack the presenting view controller would decide the dismissal logic.

(just a thought)

Reply:
I changed this to a modal presentation style, so now it's being used!